src/unix/core.c

```diff
@@ -1648,8 +1648,15 @@ int uv_thread_setpriority(uv_thread_t tid, int priority) {
   max = sched_get_priority_max(policy);
 #endif
 
-  if (min == -1 || max == -1)
-    return UV__ERR(errno);
+  /*
+   * The underlying pthread lib does not support prio
+   * There is nothing more we can do here but pretend
+   * that it worked.
+   */
+  if (min == -1 || max == -1) {
+    min = 0;
+    max = 0;
+  }
```

Comment:
I don't think we want to make such changes on platforms where we _do_ expect the call to succeed. Can I suggest using an #ifdef?

Do I read your changes correctly when I say this whole function can probably turned into a no-op on hurd?

Reply:
Thanks for your review, @bnoordhuis !

> Do I read your changes correctly when I say this whole function can probably turned into a no-op on hurd?

Yes, and I will rework the PR to do so.

I was hoping that the current approach would help other systems in the same situation, since ENOSYS is unexpected when the implementation is not a stub.

On the other hand, there is a mechanism in glibc to detect stub implementations ; there is support for this in autoconf, but I could not find something similar in CMake. It makes a specific system test the last possibility remaining.

Reply:
Thanks for your review, @bnoordhuis !

> Do I read your changes correctly when I say this whole function can probably turned into a no-op on hurd?

Yes, and I will rework the PR to do so.

I was hoping that the current approach would help other systems in the same situation, since ENOSYS is unexpected when the implementation is not a stub.

On the other hand, there is a mechanism in glibc to detect stub implementations ; there is support for this in autoconf, but I could not find something similar in CMake. It makes a specific system test the last possibility remaining.